src/api.cpp

```diff
@@ -5102,18 +5168,49 @@ clEnqueueCopyImage(cl_command_queue cq, cl_mem src_image, cl_mem dst_image,
     if (!src_img->has_same_format(dst_img)) {
         return CL_IMAGE_FORMAT_MISMATCH;
     }
+    assert(src_img->element_size() == dst_img->element_size());
 
     std::array<size_t, 3> src_orig = {src_origin[0], src_origin[1],
                                       src_origin[2]};
     std::array<size_t, 3> dst_orig = {dst_origin[0], dst_origin[1],
                                       dst_origin[2]};
     std::array<size_t, 3> reg = {region[0], region[1], region[2]};
 
-    auto cmd = std::make_unique<cvk_command_image_image_copy>(
-        command_queue, src_img, dst_img, src_orig, dst_orig, reg);
+    if (src_img->is_backed_by_buffer_view() &&
+        dst_img->is_backed_by_buffer_view()) {
+        auto cmd = new cvk_command_copy_buffer(
+            command_queue, CL_COMMAND_COPY_BUFFER,
```

Comment:
```suggestion
            command_queue, CL_COMMAND_COPY_IMAGE,
```

That's what the application should see.

Reply:
Done